### PR TITLE
Fixes #372 Add options to modify behaviour of autoindex

### LINKED ISSUE #372: Expose certain properties of the auto-generated index to the yaml file
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

- Right now, the behaviour of autogenerated index is limited in the fact that it hardcodes the fact that a file named README would be included at the top of the generated docs.
- Javadocs if present would be included at the end

To modify any aspect of this index, users have to provide their own index thereby negating the advantage of the auto-generation.

### Expected behaviour

Allow users to modify this behaviour so that the auto-generated index would be more useful.

### PR DESCRIPTION
**Pull request by pri22296**
## Description

- Added various options to control the behavior of auto-generated index. such as whether to include javadocs, subprojects etc or not.
- Also added the ability to specify the files which should be included from root. This enables the user to specify an alternate starting page instead of the README.
- Added bug fix for `github button not generated` and `doc_url not working`

## Related Issue

Fixes #372 

## Motivation and Context

In case of master projects, It is sometime desirable to have an alternate starting page instead of the README. This PR implements that and also some other options. This leads to the auto-generated index being more flexible.

## How Has This Been Tested?

Deployment: https://yaydocw.herokuapp.com

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by pri22296** [comment]
@sch00lb0y @imujjwal96 Please review

File: README.md [hunk 1]
```diff
@@ -24,6 +24,16 @@ metadata:
   projectname: Yaydoc    # Name of the Project, default: Name of the repository
   version: development   # Version of the Project, default: UTC date of latest deployment
   debug: true            # Enables detailed logging, default: false
+  autoindex:             # Control various properties of the auto-generated index
+    include:             # Documents which should be included at the top of the index
+      - README.md
```
Comment (imujjwal96) [inline README.md]: what does this do?

Comment (imujjwal96) [inline README.md]: Please explain each of the option in detail

Comment (pri22296) [inline README.md]: @imujjwal96 Updated

**Comment by imujjwal96** [comment]
@pri22296 Can you show a sample deployment? I'd suggest if you could make changes to your local fork of susi_server and generate docs from it

**Comment by pri22296** [comment]
@imujjwal96 Here is a sample deployment. https://yaydocw.herokuapp.com/preview/priyamsingh.22296@gmail.com/1a440402-104f-4b32-b7cd-ae4480e87a42_preview/

I have also made some changes where felt necessary while generating sample deployment.

You can see the source file for starting page here. https://raw.githubusercontent.com/pri22296/susi_server/development/STARTPAGE.md

@sch00lb0y I have also included the fix for CNAME issue in this PR.

**Pull request merged**